doc/user/methods/decision_tree.md

```diff
@@ -0,0 +1,135 @@
+
+| **type** | **name** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| [`arma::mat`](../matrices.md) | **`data`** | [Column-major](../matrices.md) training matrix. | _(N/A)_ |
+<!-- TODO: the linked tutorial is inappropriate, and we need a simpler
+DatasetInfo tutorial that discusses how to load and work with categorical data
+in very straightforward terms -->
+| [`data::DatasetInfo`](../../tutorials/datasetmapper.md) | **`datasetInfo`** |
+Dataset information, specifying type information for each dimension. | _(N/A)_ |
+<!-- TODO: update matrices.md to include a section on labels, and how to use
+`NormalizeLabels()` to map to [0, numClasses - 1]. -->
+| [`arma::Row<size_t>`]('../matrices.md') | **`labels`** | Training labels, between `0` and `numClasses - 1` (inclusive). | _(N/A)_ |
+<!-- TODO: include a bit about instance weights in matrices.md -->
+| [`arma::rowvec`]('../matrices.md') | **`weights`** | Weights for each training point. | _(N/A)_ |
+| `size_t` | **`numClasses`** | Number of classes in the dataset. | _(N/A)_ |
+| `size_t` | **`minimumLeafSize`** | Minimum number of points in each leaf node. | `10` |
+| `double` | **`minimumGainSplit`** | Minimum gain for a node to split. | `1e-7` |
+| `size_t` | **`maximumDepth`** | Maximum depth for the tree. (0 means no limit.) | `0` |
+
+### Training
+
+If training is not done as part of the constructor call, it can be done with the
+`Train()` member function, which has several overloads.
+
+ * `tree.Train(`_`data`_`, `_`labels`_`, `_`numClasses`_`)` (train on numerical-only data)
+ * `tree.Train(`_`data`_`, `_`labels`_`, `_`numClasses`_`, `_`minimumLeafSize`_`, `_`minimumGainSplit`_`, `_`maximumDepth`_`)` (train on numerical-only data with hyperparameters)
+
+ * `tree.Train(`_`data`_`, `_`labels`_`, `_`numClasses`_`, `_`weights`_`)` (train on weighted numerical-only data)
+ * `tree.Train(`_`data`_`, `_`labels`_`, `_`numClasses`_`, `_`weights`_`, `_`minimumLeafSize`_`, `_`minimumGainSplit`_`, `_`maximumDepth`_`)` (train on weighted numerical-only data with hyperparameters)
+
+ * `tree.Train(`_`data`_`, `_`datasetInfo`_`, `_`labels`_`, `_`numClasses`_`)` (train on mixed categorical data)
+ * `tree.Train(`_`data`_`, `_`datasetInfo`_`, `_`labels`_`, `_`numClasses`_`, `_`minimumLeafSize`_`, `_`minimumGainSplit`_`, `_`maximumDepth`_`)` (train on mixed categorical data with hyperparameters)
+
+ * `tree.Train(`_`data`_`, `_`datasetInfo`_`, `_`labels`_`, `_`numClasses`_`, `_`weights`_`)` (train on weighted mixed categorical data)
+ * `tree.Train(`_`data`_`, `_`datasetInfo`_`, `_`labels`_`, `_`numClasses`_`,
+   `_`weights`_`, `_`minimumLeafSize`_`, `_`minimumGainSplit`_`, `_`maximumDepth`_`)` (train on weighted mixed categorical data with hyperparameters)
```

Comment:
do we keep the declaration or replace them with examples directly?

Reply:
Hmm, it's a good point.  I was hoping to keep all the different overloads close together in the documentation, like for Armadillo and ensmallen.  Also, I'm not sure that we need an example for every single constructor---the difference between the first two, for instance, is trivial.  I'm trying to think about how I usually use Armadillo documentation: I take a look at the function signatures, and then take a guess at how I'd use that in my code, and usually, the guess is right.  If not, then I would go to the examples or further documentation to see if I missed something.  I'm not sure if that's how most people look at Armadillo documentation, but that's at least what I do...